bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,609 @@
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.psd");
+            } else if (FileExtensionMimeTypeConstants._SVG.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.svg");
+            } else if (FileExtensionMimeTypeConstants._TIFF_TIF.equals(mimeType)) {
+                inputStream = getCorrectBinaryTypeStream(node, "tiff", "tif");
+            } else if (FileExtensionMimeTypeConstants._TXT.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.txt");
+            } else if (FileExtensionMimeTypeConstants._WAV.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.wav");
+            } else if (FileExtensionMimeTypeConstants._WEBM.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.webm");
+            } else if (FileExtensionMimeTypeConstants._WMA.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.wma");
+            } else if (FileExtensionMimeTypeConstants._WMV.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.wmv");
+            } else if (FileExtensionMimeTypeConstants._XLS.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.xls");
+            } else if (FileExtensionMimeTypeConstants._XLSX.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.xlsx");
+            } else if (FileExtensionMimeTypeConstants._XML.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.xml");
+            } else if (FileExtensionMimeTypeConstants._ZIP.equals(mimeType)) {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.zip");
+            } else {
+                inputStream = this.dynamicClassLoaderManager.getDynamicClassLoader().getResourceAsStream("/remoteassets/remote_asset.jpeg");
+            }
+
+            node.setProperty(JcrConstants.JCR_DATA, this.valueFactory.createBinary(inputStream));
+        } finally {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Binary added for node '{}'.", node.getPath());
```

Comment:
Since we're in `finally` we dont know if it was successful or not - this should instead be under the line where we call `node.setProperty(Jcrconstants.JCR_Data...`